06_puppy/sirigithub/stores_test.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type storesSuite struct {
+	suite.Suite
+	store     Storer
+	newStorer func() Storer
+	puppy1    Puppy
+}
+
+func (suite *storesSuite) SetupTest() {
+	suite.store = suite.newStorer()
+	suite.puppy1 = Puppy{ID: 1, Breed: "cavalier", Color: "White", Value: "$1700"}
+	er := suite.store.CreatePuppy(&suite.puppy1)
+	if er != nil {
+		suite.FailNow("Test setup failed to create puppy", er)
+	}
+
+}
+func (suite *storesSuite) TestCreatePuppySuccesful() {
+	r := require.New(suite.T())
+	newPuppy := Puppy{ID: 2, Breed: "Dobberman", Color: "Black", Value: "$800"}
+	er := suite.store.CreatePuppy(&newPuppy)
+	r.NoError(er, "Create should not produce an error")
+}
+func (suite *storesSuite) TestReadPuppySuccesful() {
+	//given
+	assert := assert.New(suite.T())
+	r := require.New(suite.T())
+	//when
+	puppy, err := suite.store.ReadPuppy(suite.puppy1.ID)
+	//then
+	r.NoError(err, "Read should not produce an error")
+	assert.Equal(puppy, &suite.puppy1)
+}
+func (suite *storesSuite) TestCreatePuppyError() {
+	//given
+	assert := assert.New(suite.T())
+	//when
+	createEr := suite.store.CreatePuppy(&suite.puppy1)
+	//then
+	assert.Error(createEr, "Cannot create existing puppy")
+}
+func (suite *storesSuite) TestReadPuppyDoesNotExist() {
+	assert := assert.New(suite.T())
+	_, err := suite.store.ReadPuppy(100)
+	assert.Error(err, "Cannot read a non-existing puppy")
+}
+func (suite *storesSuite) TestUpdatePuppySuccesful() {
+	//given
+	assert := assert.New(suite.T())
+	existingPuppy := suite.puppy1
+	//when
+	existingPuppy.Color = "Brown"
+	existingPuppy.Breed = "Dobberman"
+	err := suite.store.UpdatePuppy(existingPuppy.ID, &existingPuppy)
+	//then
+	assert.NoError(err, "Update should not throw an error")
+	updatedPuppy, err := suite.store.ReadPuppy(existingPuppy.ID)
+	assert.NoError(err, "Read of updated puppy should be successful")
+	assert.NotEqual(updatedPuppy, &suite.puppy1, "Puppy values are updated")
+
+}
+func (suite *storesSuite) TestUpdatePuppyIdDoesNotExist() {
+	assert := assert.New(suite.T())
+	puppy := Puppy{ID: 10, Breed: "cavalier", Color: "White", Value: "$1700"}
+	err := suite.store.UpdatePuppy(10, &puppy)
+	assert.Error(err, "Cannot update non existing puppy")
+}
+func (suite *storesSuite) TestUpdatePuppyIdMismatch() {
+	assert := assert.New(suite.T())
+	puppy := Puppy{ID: 12, Breed: "mix", Color: "White", Value: "$1700"}
+	err := suite.store.UpdatePuppy(1, &puppy)
+	assert.Error(err, "Update puppy with different Id throws error")
```

Comment:
bad error message. Go doesn't throw errors, it returns them.
msg should be: "Update puppy with different ids doesn't return error."

Reply:
Refactored UpdatePuppy to use testify/suite and check for the error message being returned.